### PR TITLE
Notification channels

### PR DESCRIPTION
**Pull request by biafra23**
Changes proposed in this pull request:
- Add Notification Channel Group per added safe
- Update Group when local safe name is changed
- Delete Notification Channel Group when a safe is removed from the app
- On app startup: Create Notification Channel Group per existing safe (this can be removed in a future update)

Home -> Safe Multisig -> App Info -> Notifications: 
Screenshot_20210219-234453

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/ui/safe/add/AddSafeNameViewModel.kt [hunk 1]
```diff
@@ -31,7 +33,8 @@ class AddSafeNameViewModel
             runCatching {
                 val safe = Safe(address, localName.trim())
                 safeRepository.saveSafe(safe)
-                notificationRepository.registerSafe(safe.address)
+                notificationRepository.registerSafe(safe)
+                notificationManager.createNotificationChannelGroup(safe)
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/safe/add/AddSafeNameViewModel.kt:37]: the channel should also be removed when removing a safe, right?

File: app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt [hunk 2]
```diff
@@ -40,21 +50,57 @@ class NotificationManager(
             }
         }
 
+    fun createNotificationChannelGroup(safe: Safe) {
+        val id = safe.address.asEthereumAddressChecksumString()
+        val name = safe.localName
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            notificationManager.createNotificationChannelGroup(NotificationChannelGroup(id, name))
+            createNotificationChannel(id)
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt:58]: should notification channel id's have also some common prefix?

File: app/src/main/java/io/gnosis/safe/ui/safe/add/AddSafeNameViewModel.kt (hunk 1 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/ui/safe/add/AddSafeNameViewModel.kt:37]: `deleteNotificationChannelGroup()` removes the all groups channels and the group itself. 
```
    fun deleteNotificationChannelGroup(safe: Safe) {
        val subChannels = ChannelType.values()

        subChannels.forEach { subChannel ->
            notificationManager.deleteNotificationChannel(safe.address.asEthereumAddressChecksumString() + "." + context.getString(subChannel.resId))
        }
        notificationManager.deleteNotificationChannelGroup(safe.address.asEthereumAddressChecksumString())
    }
´´´

Called in `SafeSettingsViewModel.removeSafe()`

File: app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt (hunk 2 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt:58]: It's not necessary. What would be an advantage?

File: app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt [hunk 3]
```diff
@@ -1,35 +1,45 @@
...
+    private val balanceFormatter: BalanceFormatter,
+    private val safeRepository: SafeRepository,
+    private val notificationManager: NotificationManagerCompat = NotificationManagerCompat.from(context)
 ) {
 
-    private val notificationManager = NotificationManagerCompat.from(context)
-
     init {
-        createNotificationChannel(CHANNEL_ID)
+        if (Build.VERSION.SDK_INT >= 26) {
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt:34]: where is the else branch? What about skd levels < 26? Right not not channel will be created at all

File: app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt [hunk 4]
```diff
@@ -1,35 +1,45 @@
...
+            // For upgrading users. Can be removed in a future release?
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt:35]: can we check if channel group already exists?

File: app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt (hunk 3 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt:34]: The previous code also didn't create a channel for api level < 26.

File: app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt (hunk 4 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt:35]: Apparently, we can. Will rewrite to check first.

File: app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt (hunk 3 above)
Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt:34]: you are right, there simply were no different channels at those api levels

**Review by elgatovital — APPROVED** [review]

**Pull request merged**